src/jit/hwintrinsicxarch.cpp

```diff
@@ -27,180 +15,166 @@ static const HWIntrinsicInfo hwIntrinsicInfoArray[] = {
 #include "hwintrinsiclistxarch.h"
 };
 
-extern const char* getHWIntrinsicName(NamedIntrinsic intrinsic)
+//------------------------------------------------------------------------
+// lookup: Gets the HWIntrinsicInfo associated with a given NamedIntrinsic
+//
+// Arguments:
+//    id -- The NamedIntrinsic associated with the HWIntrinsic to lookup
+//
+// Return Value:
+//    The HWIntrinsicInfo associated with id
+const HWIntrinsicInfo& HWIntrinsicInfo::lookup(NamedIntrinsic id)
```

Comment:
It seems like some of this could be shared, e.g. in a hwintrinsic.cpp or hwintrinsicshared.cpp

Reply:
Yes, I agree.

Reply:
Will log a work item tracking this.